src/mscorlib/src/System/Array.cs

```diff
@@ -45,7 +45,7 @@ public abstract class Array : ICloneable, IList, IStructuralComparable, IStructu
         [ReliabilityContract(Consistency.WillNotCorruptState, Cer.MayFail)]
         public static void Resize<T>(ref T[] array, int newSize) {
             if (newSize < 0)
-                throw new ArgumentOutOfRangeException("newSize", Environment.GetResourceString("ArgumentOutOfRange_NeedNonNegNum"));
+                ThrowArgumentOutOfRangeNeedNonNegNum(nameof(newSize));
```

Comment:
Use of constant strings results in an additional call, that's why ThrowHelper passes the argument as an enum instead.